tests/test_tomography_gate_set_tomography.py

```diff
@@ -0,0 +1,401 @@
+    else:
+        prepared_gates = _measurement_basis(j, nqubits)
+        for groundtruth, gate in zip(correct_gates[nqubits][j], prepared_gates):
+            _compare_gates(groundtruth, gate)
+            for g1, g2 in zip(groundtruth.basis, gate.basis):
+                _compare_gates(g1, g2)
+
+
+@pytest.mark.parametrize(
+    "j,nqubits",
+    INDEX_NQUBITS,
+)
+def test__get_observable(j, nqubits):
+    correct_observables = {
+        1: [
+            (S(1),),
+            (qibo.symbols.Z(0),),
+            (qibo.symbols.Z(0),),
+            (qibo.symbols.Z(0),),
+        ],
+        2: [
+            (S(1), S(1)),
+            (S(1), qibo.symbols.Z(1)),
+            (S(1), qibo.symbols.Z(1)),
+            (S(1), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), S(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), S(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), S(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+        ],
+    }
+    correct_observables[1] = [
+        SymbolicHamiltonian(h[0]).form for h in correct_observables[1]
+    ]
+    correct_observables[2] = [
+        SymbolicHamiltonian(reduce(lambda x, y: x * y, h)).form
+        for h in correct_observables[2]
+    ]
+    errors = {(0, 3): ValueError, (17, 1): IndexError}
+    if (j, nqubits) in [(0, 3), (17, 1)]:
+        with pytest.raises(errors[(j, nqubits)]):
+            prepared_observable = _get_observable(j, nqubits)
+    else:
+        prepared_observable = _get_observable(j, nqubits).form
+        groundtruth = correct_observables[nqubits][j]
+        assert groundtruth == prepared_observable
+
+
+@pytest.mark.parametrize("nqubits", range(3, 7))
+def test_expectation_value(backend, nqubits):
+
+    if nqubits == 3:
+        test_circuit = qibo.models.Circuit(nqubits)
+        test_circuit.add(gates.TOFFOLI(0, 1, 2))
+        test_circuit.add(gates.M(*np.arange(0, nqubits, 1)))
+
+    else:
+        RandUnitary = RU(2**nqubits)
+        test_circuit = qibo.models.Circuit(nqubits)
+        test_circuit.add(gates.Unitary(RandUnitary, *np.arange(0, nqubits, 1)))
+        test_circuit.add(gates.M(*np.arange(0, nqubits, 1)))
+    k = 1
+    j = 1
+    with pytest.raises(ValueError):
+        expectation_val = _expectation_value(
+            test_circuit,
+            j,
+            nshots=int(1e4),
+            backend=backend,
+        )
+
+
+@pytest.mark.parametrize("nqubits", range(1, 4))
+@pytest.mark.parametrize("gate", [gates.CNOT(0, 1), gates.TOFFOLI(0, 1, 2), None])
+def test_gate_set_tomography(backend, nqubits, gate):
```

Comment:
```suggestion
@pytest.mark.parametrize("nqubits,gate", [(1, gates.CNOT(0,1)), (3, gates.TOFFOLI(0, 1, 2))])
def test_gate_set_tomography_value_error(backend, nqubits, gate):
```
I don't think you need to test any other combination, since you're actually only checking that the `ValueError` is raised